src/benchcab/utils/repo.py

```diff
@@ -137,8 +137,11 @@ def __init__(
         """
         self.url = url
         self.branch = branch
+        is_bare_directory = realisation_path.is_dir() and (
+            not (realisation_path / ".git").exists()
+        )
```

Comment:
I have one reservation. If the path is a directory but does not contain a .git path, does that mean there is something going really wrong with benchcab? And is this implementation hiding the issue if that arises? I'm not sure this should be the point in the code to check for that but I'm worried about hiding potential integration issues. 

Or I'm simply misremembering what the code does anyway...

Obviously, the same applies to the SVN case.

Any thoughts?

Reply:
> If the path is a directory but does not contain a .git path, does that mean there is something going really wrong with benchcab?

It doesn't mean things have gone wrong (this condition is true when invoking `benchcab checkout` in a fresh work directory). I suspect the complexity I've introduced is due to how I designed the `Repo` class and the `create_repo` factory function in https://github.com/CABLE-LSM/benchcab/pull/208. It works but it's not pretty.

Reply:
OK. All good then.